cinema/serializers.py

```diff
@@ -1,24 +1,83 @@
 from rest_framework import serializers
 
-from cinema.models import Movie
+from cinema.models import Movie, Actor, Genre, CinemaHall
 
 
-class MovieSerializer(serializers.Serializer):
-    id = serializers.IntegerField(read_only=True)
-    title = serializers.CharField(max_length=255)
-    description = serializers.CharField()
-    duration = serializers.IntegerField()
+class MovieSerializer(serializers.ModelSerializer):
+    actors = serializers.PrimaryKeyRelatedField(many=True,
+                                                queryset=Actor.objects.all())
+    genres = serializers.PrimaryKeyRelatedField(many=True,
+                                                queryset=Genre.objects.all())
+
+    class Meta:
+        model = Movie
+        fields = ['id', 'title', "duration", "actors", "genres"]
 
     def create(self, validated_data):
-        return Movie.objects.create(**validated_data)
+        actors = validated_data.pop("actors")
+        genres = validated_data.pop("genres")
+        movies = Movie.objects.create(**validated_data)
+        movies.set(actors)
+        movies.set(genres)
+        return movies
 
     def update(self, instance, validated_data):
+        actors = validated_data.pop("actors")
+        genres = validated_data.pop("genres")
         instance.title = validated_data.get("title", instance.title)
         instance.description = validated_data.get(
             "description", instance.description
         )
         instance.duration = validated_data.get("duration", instance.duration)
+        instance.genres = validated_data.get("genres", instance.genres)
```

Comment:
In the update method, directly assigning ManyToMany fields like 'instance.genres = ...' is incorrect. Instead, use 'instance.genres.set(genres)' to update the ManyToMany relationships.